wallet/src/store/utxo.rs

```diff
@@ -136,18 +172,22 @@ impl<KEY: Groupable> UtxoStore<KEY> {
     ///
     /// this allows optimizing the search of inputs and to favors using
     /// inputs of the same key to preserve privacy
-    pub fn groups(&self) -> impl Iterator<Item = std::cell::Ref<'_, UtxoGroup<KEY>>> {
-        self.by_derivation_path.values().map(|r| r.borrow())
+    pub fn groups(&self) -> impl Iterator<Item = &GroupRef<K>> {
+        self.by_derivation_path.iter().map(|(_, v)| v)
     }
 
     /// get the UTxO, not grouped, ordered by value
     pub fn utxos(&self) -> impl Iterator<Item = &UTxO> {
-        self.by_value.values().flatten()
+        self.by_value
```

Comment:
@NicolasDP utxos are sorted here per call instead of the data structure. It could be less performant but still should not break.
cc : @enzoc4